src/ApplicationInsights.Kubernetes/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -62,45 +58,59 @@ static ApplicationInsightsExtensions()
+
             // 2 minutes by default maximum to wait for spinning up the container.
             timeout = timeout ?? TimeSpan.FromMinutes(2);
 
-            // According to the code, adding logging will not overwrite existing logging classes
-            // https://github.com/aspnet/Logging/blob/c821494678a30c323174bea8056f43b93a3ca6f4/src/Microsoft.Extensions.Logging/LoggingServiceCollectionExtensions.cs
-            // Becuase it uses 'TryAdd()' extenion method on service collection.
-            serviceCollection.AddLogging();
-
-            _logger.LogInformation(Invariant($"ApplicationInsights.Kubernetes.Version:{SDKVersionUtils.Instance.CurrentSDKVersion}"));
+            logger.LogInformation(Invariant($"ApplicationInsights.Kubernetes.Version:{SDKVersionUtils.Instance.CurrentSDKVersion}"));
             try
             {
-                serviceCollection = BuildK8sServiceCollection(serviceCollection, timeout.Value, kubernetesServiceCollectionBuilder);
+                serviceCollection = BuildK8sServiceCollection(serviceCollection, timeout.Value, k8sDetector, logger, kubernetesServiceCollectionBuilder);
             }
             catch (Exception ex)
             {
-                _logger.LogError("Failed to fetch ApplicaitonInsights.Kubernetes' info. Details " + ex.ToString());
+                logger.LogError("Failed to fetch ApplicaitonInsights.Kubernetes' info. Details " + ex.ToString());
             }
             return serviceCollection;
         }
 
         private static IServiceCollection BuildK8sServiceCollection(
             IServiceCollection services,
             TimeSpan timeout,
+            K8sDetector k8sDetector,
+            ILogger<KubernetesServiceCollectionBuilder> logger,
             IKubernetesServiceCollectionBuilder kubernetesServiceCollectionBuilder = null)
         {
-            kubernetesServiceCollectionBuilder = kubernetesServiceCollectionBuilder ?? new KubernetesServiceCollectionBuilder();
+            k8sDetector = k8sDetector ?? K8sDetectorFactory.Instance.CreateDetector();
+            kubernetesServiceCollectionBuilder = kubernetesServiceCollectionBuilder ?? new KubernetesServiceCollectionBuilder(k8sDetector, logger);
             services = kubernetesServiceCollectionBuilder.InjectServices(services, timeout);
             return services;
         }
+
+        /// <summary>
+        /// Gets a logger for given type.
+        /// Note: This method leads to build service provider during the injection of services and shall be avoid whenever possible.
+        /// </summary>
+        private static ILogger<T> GetLogger<T>(this IServiceCollection services)
+        {
+            // According to the code, adding logging will not overwrite existing logging classes
+            // https://github.com/aspnet/Logging/blob/c821494678a30c323174bea8056f43b93a3ca6f4/src/Microsoft.Extensions.Logging/LoggingServiceCollectionExtensions.cs
```

Comment:
Update reference so that it points to master branch, not some specific commit

Reply:
On github, when one clicks on the specific line and use 'Copy permalink', it will generate the path like that. My guess is that is the way to making sure the reference exists.

Reply:
This is a link that refers to the version of the code as it was after a specific commit, namely, https://github.com/aspnet/Logging/commit/c821494678a30c323174bea8056f43b93a3ca6f4 

My point is, if you are making assumptions about how something is implemented, you should be willing to point to the latest version of the code, and not some implementation from the past, which might have changed since.